pyresample/test/test_gradient.py

```diff
@@ -599,7 +599,7 @@ def test_check_overlap():
     assert check_overlap(poly1, poly2) is False
 
 
-@mock.patch('pyresample.gradient.get_geostationary_bounding_box')
+@mock.patch('pyresample.gradient.get_geostationary_bounding_box_in_lonlats')
```

Comment:
I really don't like how all these old tests mock the functions being called. I know this isn't what this PR is about, but mocking the bounding box function called by the border function in this test, what's being tested then? The same for the test below it, lots of mocks. In the future it'd be nice if we could construct an area definition that just produced the results we wanted.

Reply:
I agree. Let's aim to keep mocking at a minimum.